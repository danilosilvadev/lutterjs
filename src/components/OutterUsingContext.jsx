import React from 'react';
import { mapContextToProps } from '../rxreactjs';

function OutterUsingContext(props) {
  const { state, actions } = props;
  return (
    <div>
      <div>
        {state && state.someData}
      </div>
      <button onClick={() => actions.changeSomeData()}>
        Change the data
      </button>
    </div>
  )
}

// This is how we inject it
export default mapContextToProps(OutterUsingContext, 'MY_CONTEXT');
