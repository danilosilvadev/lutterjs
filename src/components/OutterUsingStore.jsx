import React from 'react';
import { mapStoreToProps } from '../rxreactjs';

const handleSomeData = (store) => (
  store && store.MY_CONTEXT && store.MY_CONTEXT.state.someData
);

function OutterUsingStore(props) {
  const { store } = props;
  return (
    <div>
      <div>
        {handleSomeData(store)}
      </div>
      <button onClick={() => store.MY_CONTEXT.actions.changeSomeData()}>
        Change the data
      </button>
    </div>
  )
}

// This is how we inject it
export default mapStoreToProps(OutterUsingStore);
