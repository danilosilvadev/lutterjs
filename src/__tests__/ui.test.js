import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import OutterUsingStore from '../components/OutterUsingStore';
import OutterUsingContext from '../components/OutterUsingContext';
import Provider from '../components/provider';

describe('UI tests', () => {
  const wrapperProvider = shallow(
    <Provider />
  );

  it('should render Provider correctly', () => {
  const tree = renderer
      .create(wrapperProvider)
      .toJSON();
  expect(tree).toMatchSnapshot();
  });

  const wrapperStore = shallow(
    <OutterUsingStore />
  );

  it('should render OutterUsingStore correctly', () => {
  const tree = renderer
      .create(OutterUsingStore)
      .toJSON();
  expect(tree).toMatchSnapshot();
  });

  const wrapperContext = shallow(
    <OutterUsingContext />
  );

  it('should render OutterUsingContext correctly', () => {
  const tree = renderer
      .create(OutterUsingContext)
      .toJSON();
  expect(tree).toMatchSnapshot();
  });
});
