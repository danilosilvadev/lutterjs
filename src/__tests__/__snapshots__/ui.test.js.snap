// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI tests should render OutterUsingContext correctly 1`] = `null`;

exports[`UI tests should render OutterUsingStore correctly 1`] = `null`;

exports[`UI tests should render Provider correctly 1`] = `
Array [
  <div>
    <div>
        A data coming from fetching
    </div>
    <button
        onClick={[Function]}
    >
        Change the data
    </button>
</div>,
  <div>
    <div>
        A data coming from fetching
    </div>
    <button
        onClick={[Function]}
    >
        Change the data
    </button>
</div>,
]
`;
